File: 프로그래머스/lv1/12943. 콜라츠 추측/콜라츠 추측.py
def solution(num):
    if num == 1:
        return 0

    answer = 0

    while True:
        answer += 1
        if not (num % 2):
            num //= 2
        else:
            num = num * 3 + 1
        if num == 1:
            break
        if answer == 500:
            return -1

    return answer

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.cpp
using namespace std;

int solution(int num) {
    if (num == 1) return 0;
    
    long long n = num;
    int answer = 0;
    
    while (true) {
        answer++;
        
        if (n % 2 == 0) {
            n /= 2;
        } else {
            n = n * 3 + 1;
        }
        
        if (n == 1) break;
        
        if (answer == 500) return -1;
    }

    return answer;
}